src/utils/exportUtils.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  TDExportBackground,
+  TDExportType,
+  TDShape,
+  TDShapeType,
+  TLDR,
+  TldrawApp,
+} from "@tldraw/tldraw";
+import { Utils } from "@tldraw/core";
+
+export const getBlob = async (
+  app: TldrawApp,
+  type: string,
+): Promise<Blob | undefined> => {
+  const format = type as TDExportType;
+  if (format === TDExportType.JSON) return;
+
+  const svg = await getSvg(app, format);
+  if (!svg) return;
+
+  if (format === TDExportType.SVG) {
+    const svgString = TLDR.getSvgString(svg, 1);
+    const blob = new Blob([svgString], { type: "image/svg+xml" });
+
+    return blob;
+  }
+
+  const imageBlob = await TLDR.getImageForSvg(svg, format, {
+    scale: 2,
+    quality: 1,
+  });
+
+  return imageBlob;
+};
+
+const getSvg = async (
```

Comment:
try to split this implementation into smaller functions

Reply:
done